src/parser/tokenizer.ts

```diff
@@ -207,6 +207,14 @@ export class Tokenizer {
     const value = this.readQuoted() || this.readRange()
     if (value) return value
 
+    if (this.peek() === '[') {
+      this.p++
+      const prop = this.readQuoted()
+      if (!prop) return
+      this.readTo(']')
+      return new PropertyAccessToken(prop, [], this.p)
+    }
+
```

Comment:
How about `["foo"]["bar"]` or `["foo"].bar`?

We need adapt the following logic (the `PropertyAccessToken` part) to support this case (remove the `variable` constructor argument and make it the first props maybe?).

Reply:
Makes sense, although the number of potential valid syntaxes are getting a bit scary. Do you think that could be postponed into the future and this feature shipped as is (once I fix the bugs)?

Reply:
> the number of potential valid syntaxes are getting a bit scary

Seems it can be done by consume the `[`, call readValue recursively(as in props brackets), and consume the `]`. This way the following cases will be supported also:

```liquid
{{["foo"].bar}}
{{[0].bar}}
{{[key].bar}}
```